solved_ATM_ex/solved_ATM_ex.py

```diff
@@ -0,0 +1,180 @@
+    not_finish_actions = True
+
+    while not_finish_actions:
+        if costumer_indent in costumers_dictionary and \
+                costumers_dictionary[costumer_indent][
+                    COSTUMER_PASSWORD_PLACE_IN_DICTIONARY] \
+                == costumer_password:
+
+            try:
+                print("For checking your balance press 1")
+                print("For cash withdrawal press 2")
+                print("For cash deposit press 3")
+                print("For change password press 4")
+                print("To exit press -1")
+                costumer_choice = int(input("Pressing: \n"))
+
+                if costumer_choice == 1:
+                    print(f"Your balance: "
+                          f"{float(costumers_dictionary[costumer_indent][COSTUMER_BALANCE_PLACE_IN_DICTIONARY])}\n")
+
+                elif costumer_choice == 2:
+                    amount_to_withdrawal = float(input("Please enter amount "
+                                               "you want to withdrawal: \n"))
+
+                    if cash_withdrawal(costumer_indent, amount_to_withdrawal,
+                                       costumers_dictionary):
+                        print("The action was succeeded\n")
+
+                    else:
+                        print("Failed - check your amount\n")
+
+                elif costumer_choice == 3:
+                    amount_to_deposit = float(input("Please enter amount you "
+                                                    "want to deposit: \n"))
+
+                    if cash_deposit(costumer_indent, amount_to_deposit,
+                                    costumers_dictionary):
+                        print("The action was succeeded\n")
+
+                    else:
+                        print("Failed - check your amount\n")
+
+                elif costumer_choice == 4:
+                    new_password = input("Enter a new password: \n")
+                    if change_password(costumer_indent, new_password,
+                                       costumers_dictionary):
+                        print("Password changed")
+
+                    else:
+                        print("Not a good password, needs to be 4 digits:")
+
+                elif costumer_choice == -1:
+                    not_finish_actions = False
+
+                else:
+                    print("Choose 1/2/3/4/-1 option: ")
+
+            except ValueError:
+                print("Value Error! your input needs to be digit value\n")
+
+        else:
+            print("Your ID or your password is not correct. Please try to "
+                  "enter again:\n")
+            costumer_indent = input("Enter your id: ")
+            costumer_password = input("Enter your password: ")
+
+        with open(sys.argv[ATM_FILE], 'w') as atm_file:
+            for line in costumers_dictionary:
+                atm_file.write(
+                    f"{line} {' '.join(costumers_dictionary[line])}\n")
```

Comment:
The whole reason to use fstrings is that they allow you to write clean readable strings. You don't need to use join in there. Just having a space between your brackets means that there is a space between them.

You should make sure to check your outputs since from the code you have here it looks like you are adding two spaces instead of one which is the kind of thing you should try to catch by looking at the outputs.